File: C# ad y ad/2_ Orientado a objetos/S04.2_ Calculadora/Solucion_calculadora/Calculadora/Calculadora.cs
using System;


namespace Calculadora
{
	static class Calculadora
	{
		static int cuentasHechasComun = 0;
		static int cuentasHechasCientifica = 0;

		static TipoDeCalculadora tipoDeCalculadora = TipoDeCalculadora.Comun;


		public static float Sumar(float numero1, float numero2)
		{
			cuentasHechasComun++;

			return numero1 + numero2;
		}
		public static float Restar(float numero1, float numero2)
		{
			cuentasHechasComun++;

			return numero1 - numero2;
		}
		public static float Multiplicar(float numero1, float numero2)
		{
			cuentasHechasComun++;

			return numero1 * numero2;
		}
		public static float Dividir(float numero1, float numero2)
		{
			cuentasHechasComun++;

			return numero1 / numero2;
		}

		public static float ObtenerPotencia(float numeroAElevar, float aCuantoElevar)
		{
			cuentasHechasCientifica++;

			return (float)Math.Pow(numeroAElevar, aCuantoElevar);
		}
		public static float ObtenerRaiz(float numeroAEstablecerRaiz, float tipoDeRaiz)
		{
			cuentasHechasCientifica++;

			return (float)Math.Pow(numeroAEstablecerRaiz, 1.0f / tipoDeRaiz);
		}

		public static void CambiarTipoDeCalculadora()
		{
			if (tipoDeCalculadora == TipoDeCalculadora.Comun)
			{
				tipoDeCalculadora = TipoDeCalculadora.Cientifica;
			}
			else
			{
				tipoDeCalculadora = TipoDeCalculadora.Comun;
			}
		}
		public static void RestablecerEstadisticas()
		{
			cuentasHechasComun = 0;
			cuentasHechasCientifica = 0;
		}

		public static int GetCuentasHechasComun()
		{
			return cuentasHechasComun;
		}
		public static int GetCuentasHechasCientifica()
		{
			return cuentasHechasCientifica;
		}
		public static TipoDeCalculadora GetTipoDeCalculadora()
		{
			return tipoDeCalculadora;
		}
	}
}

File: C++ ad y ad/3_ Orientado a objetos/S04.2_ Calculadora/Solucion_calculadora/Calculadora/calculadora.cpp
#include "calculadora.h"

#include <cmath>


int Calculadora::cuentasHechasComun = 0;
int Calculadora::cuentasHechasCientifica = 0;

TIPO_DE_CALCULADORA Calculadora::tipoDeCalculadora = TIPO_DE_CALCULADORA::COMUN;


float Calculadora::sumar(float numero1, float numero2)
{
	cuentasHechasComun++;

	return numero1 + numero2;
}
float Calculadora::restar(float numero1, float numero2)
{
	cuentasHechasComun++;

	return numero1 - numero2;
}
float Calculadora::multiplicar(float numero1, float numero2)
{
	cuentasHechasComun++;

	return numero1 * numero2;
}
float Calculadora::dividir(float numero1, float numero2)
{
	cuentasHechasComun++;

	return numero1 / numero2;
}

float Calculadora::obtenerPotencia(float numeroAElevar, int aCuantoElevar)
{
	cuentasHechasCientifica++;

	return pow(numeroAElevar, (float)aCuantoElevar);
}
float Calculadora::obtenerRaiz(float numeroAEstablecerRaiz, int tipoDeRaiz)
{
	cuentasHechasCientifica++;

	return pow(numeroAEstablecerRaiz, 1.0f / (float)tipoDeRaiz);
}

void Calculadora::cambiarTipoDeCalculadora()
{
	if (tipoDeCalculadora == TIPO_DE_CALCULADORA::COMUN)
	{
		tipoDeCalculadora = TIPO_DE_CALCULADORA::CIENTIFICA;
	}
	else
	{
		tipoDeCalculadora = TIPO_DE_CALCULADORA::COMUN;
	}
}
void Calculadora::restablecerEstadisticas()
{
	cuentasHechasComun = 0;
	cuentasHechasCientifica = 0;
}

int Calculadora::getCuentasHechasComun()
{
	return cuentasHechasComun;
}
int Calculadora::getCuentasHechasCientifica()
{
	return cuentasHechasCientifica;
}
TIPO_DE_CALCULADORA Calculadora::getTipoDeCalculadora()
{
	return tipoDeCalculadora;
}
